### PR TITLE
remove MarshalJSON/UnmarshalJSON from delta

### PR DESCRIPTION
**Pull request by pattyshack**

### CODE REVIEW

**Review by SaveTheRbtz — APPROVED** [review]
be careful with that since it might break the code that tries to feed register entries into delta trough `UnmarshalJSON` -- if there is any.

File: engine/execution/state/delta/delta.go [hunk 1]
```diff
@@ -95,30 +92,3 @@ func (d Delta) RegisterIDs() []flow.RegisterID {
...
-
-func (d Delta) MarshalJSON() ([]byte, error) {
-	m := make(flow.RegisterEntries, len(d.Data))
-	for key, value := range d.Data {
-		m = append(m, flow.RegisterEntry{Key: key, Value: value})
-	}
-	return json.Marshal(m)
-}
-
-func (d *Delta) UnmarshalJSON(data []byte) error {
```
Comment (janezpodhostnik) [inline engine/execution/state/delta/delta.go:107]: was this never unmarshalled? Interesting.

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
yeah, it's weird.  it looks like we only use marshal for the cmd line exporter, and never unmarshal.  The exporter's marshalling code is pretty weird too.  the fields are individually marshal / repacked.  I don't know why we don't just marshal the snapshot object

**Pull request merged**